playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -257,19 +259,25 @@ def content_filter_func(recording_str):
         else:
             id_prefixes = ['{}/'.format(category)]
 
-        for id_prefix in id_prefixes:
-            for key in self._s3_facade.iter_keys(
-                    prefix=self.METADATA_KEY.format(
-                        key_prefix=self.key_prefix, id=id_prefix
-                    ),
-                    start_date=start_date,
-                    end_date=end_date,
-                    content_filter=content_filter,
-                    limit=limit):
+        id_prefix_iterators = [self._s3_facade.iter_keys(
+            prefix=self.METADATA_KEY.format(
+                key_prefix=self.key_prefix, id=id_prefix
+            ),
+            start_date=start_date,
+            end_date=end_date,
+            content_filter=content_filter) for id_prefix in id_prefixes]
```

Comment:
You should also pass limit here as you for sure don't need more than the limit per day.
It will tell the underlying iterator to fetch less items

Reply:
done